src/components/EditCoursePage/index.jsx

```diff
@@ -89,32 +85,35 @@ class EditCoursePage extends React.Component {
         },
       },
     } = this.props;
+    let updatedCourseEditData = jsonDeepCopy(courseData); // eslint-disable-line prefer-const
 
     // We don't need to send the course runs as part of courseData because they are
     // pulled out and sent to their own endpoint for saving.
-    delete courseData.course_runs;
-    courseData.subjects = [
-      courseData.subjectPrimary,
-      courseData.subjectSecondary,
-      courseData.subjectTertiary,
+    delete updatedCourseEditData.course_runs;
+
+    updatedCourseEditData.subjects = [
+      updatedCourseEditData.subjectPrimary,
+      updatedCourseEditData.subjectSecondary,
+      updatedCourseEditData.subjectTertiary,
     ].filter(subject => !!subject);
-    courseData.entitlements = entitlements && entitlements[0] && [{
-      mode: courseData.mode,
-      price: courseData.price,
+    updatedCourseEditData.entitlements = entitlements && entitlements[0] && [{
+      mode: updatedCourseEditData.mode,
+      price: updatedCourseEditData.price,
       sku: entitlements[0].sku,
     }];
-    courseData.image = courseData.imageSrc;
+    updatedCourseEditData.image = updatedCourseEditData.imageSrc;
     // We are removing imageSrc so we don't send the image data twice
-    delete courseData.imageSrc;
-    courseData.video = { src: courseData.videoSrc };
-    courseData.key = key;
-    courseData.uuid = uuid;
-    /* eslint-enable no-param-reassign */
+    delete updatedCourseEditData.imageSrc;
```

Comment:
If we _really_ care about minimal data, you can also delete 
- `updatedCourseEditData.subjectPrimary`
- `updatedCourseEditData.subjectSecondary`
- `updatedCourseEditData.subjectTertiary`
- `updatedCourseEditData.videoSrc`
- `updatedCourseEditData.price`
- `updatedCourseEditData.mode`